src/lib.rs

```diff
@@ -230,50 +349,65 @@ pub fn run(args: Args) {
                 matched.store(true, Ordering::SeqCst);
             }
             buffer_writer.print(printer.get_mut()).unwrap();
+
+            Ok(())
         },
-    );
+    )?;
 
-    if !messages::errored() {
+    let mut non_fatal_errors = Arc::into_inner(non_fatal_errors)
+        .unwrap()
```

Comment:
Interesting looks like CI is failing on this invariant not being true (that I guess there's only one `Arc` reference to this `non_fatal_errors` at this point?)

Hmm I'm not sure why that's the case

But probably doesn't "have to be an issue" (eg per other comments I could avoid needing to do this "filter out the ones that aren't actual errors" here and then could just clone the `Arc`)